implementation/src/main/java/io/smallrye/jwt/auth/principal/KeyLocationResolver.java

```diff
@@ -104,7 +111,7 @@ private PublicKey tryAsJWK(JsonWebSignature jws) throws UnresolvableKeyException
         String kid = jws.getHeaders().getStringHeaderValue(JsonWebKey.KEY_ID_PARAMETER);
```

Comment:
Please move the `kid` extraction code into its own function and remove `tryAsJWK` as suggested above